lib/common/http2_client.go

```diff
@@ -11,12 +12,23 @@ import (
 	"golang.org/x/net/http2"
 )
 
+type Doer interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+type BackoffStrategy = pester.BackoffStrategy
+
+type RetrySetting struct {
+	MaxRetries  int
+	Concurrency int
+	Backoff     BackoffStrategy
+}
+
 type HTTP2Client struct {
-	client    http.Client
+	client    Doer
 	transport *http.Transport
 }
 
-func NewHTTP2Client(timeout, idleTimeout time.Duration, keepAlive bool) (client *HTTP2Client, err error) {
+func NewHTTP2Client(timeout, idleTimeout time.Duration, keepAlive bool, retrySetting *RetrySetting) (client *HTTP2Client, err error) {
```

Comment:
Given the amount of modification necessary, I'd suggest to provide another overload. E.g. leave `NewHTTP2Client` with the current interface, and make it forward to `NewPersistentHTTP2Client` with `nil`